src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs

```diff
@@ -35,54 +38,90 @@ internal class CodeActionsHandler : AbstractRequestHandler<LSP.CodeActionParams,
+            codeActions = codeActions.Where(c => !(c.Key is CodeActionWithOptions));
 
-            var result = new ArrayBuilder<LSP.SumType<LSP.Command, LSP.CodeAction>>();
-            foreach (var codeAction in codeActions)
-            {
-                // Always return the Command instead of a precalculated set of workspace edits. 
-                // The edits will be calculated when the code action is either previewed or 
-                // invoked.
-
-                // It's safe for the client to pass back the range/filename in the command to run
-                // on the server because the client will always re-issue a get code actions request
-                // before invoking a preview or running the command on the server.
+            return GetVSCodeActions(request, codeActions);
 
-                result.Add(
-                    new LSP.Command
+            // Local functions
+            static VSCodeAction[] GetVSCodeActions(
+                CodeActionParams request,
+                IEnumerable<KeyValuePair<CodeAction, CodeActionKind>> codeActions)
+            {
+                var suppressionActions = codeActions.Where(
+                    a => a.Key is AbstractConfigurationActionWithNestedActions &&
+                    (a.Key as AbstractConfigurationActionWithNestedActions)?.IsBulkConfigurationAction == false);
+
+                using var _ = ArrayBuilder<VSCodeAction>.GetInstance(out var results);
+                foreach (var codeAction in codeActions)
+                {
+                    // Temporarily filter out suppress and configure code actions, as we'll later combine them under a top-level
+                    // code action.
+                    if (codeAction.Key is AbstractConfigurationActionWithNestedActions)
                     {
-                        CommandIdentifier = RunCodeActionCommandName,
-                        Title = codeAction.Title,
-                        Arguments = new object[]
-                        {
-                                new RunCodeActionParams
-                                {
-                                    CodeActionParams = request,
-                                    Title = codeAction.Title
-                                }
-                        }
-                    });
+                        continue;
+                    }
+
+                    results.Add(GenerateVSCodeAction(request, codeAction.Key, codeAction.Value));
+                }
+
+                // Special case (also dealt with specially in local Roslyn): 
+                // If we have configure/suppress code actions, combine them under one top-level code action.
+                var configureSuppressActions = codeActions.Where(a => a.Key is AbstractConfigurationActionWithNestedActions);
+                if (configureSuppressActions.Any())
+                {
+                    results.Add(GenerateVSCodeAction(request, new CodeActionWithNestedActions(
```

Comment:
The configure and suppress issues is a code action that is dealt with specially in local Roslyn.
We can't reuse the resource string since it lies in a different layer.

Reply:
Can we move the existing string rather than duplicate?

Reply:
I think we could, although the original location makes more sense to me for the local scenario.
@mavasani, is there any issue with moving the "Configure or Suppress issues" string to CodeFixesResources instead of EditorFeaturesWpfResources so that we can use it in the LSP layer?